### PR TITLE
made plugin pipeline ready by implementing SimpleBuildStep. Note that…

### PR DESCRIPTION
**Pull request by AndreasR92**
Hi,
I made this plugin pipeline ready by implementing SimpleBuildStep. Note that this implementation duplicates some functions in order to make the plugin in work for both, freestyle and pipeline jobs.

If you have any questions feel free to contact me.

Thanks and best regards
Andy

### CODE REVIEW

**Comment by Brantone** [comment]
Checks failed.

Also concerning that a fair number of aspects are outright copied with inconspicuous deltas. In some cases, params could just be casted ... and in others, can be wrapped and call existing function.
Too much duplicate.

**Comment by AndreasR92** [comment]
Hi,
thanks for your answer.

Can you give me a hint on how to make the checks run? How can I run the existing tests locally?
Would this change be interesting for you, if I refactored my „not so nice“ first version?

Thanks
Andreas


Von: Brenton B <notifications@github.com>
Antworten an: jenkinsci/hockeyapp-plugin <reply@reply.github.com>
Datum: Mittwoch, 20. Juli 2016 um 21:11
An: jenkinsci/hockeyapp-plugin <hockeyapp-plugin@noreply.github.com>
Cc: Andreas Reichel <andreas.reichel@sovanta.com>, Author <author@noreply.github.com>
Betreff: Re: [jenkinsci/hockeyapp-plugin] made plugin pipeline ready by implementing SimpleBuildStep. Note that… (#31)


Checks failed.

Also concerning that a fair number of aspects are outright copied with inconspicuous deltas. In some cases, params could just be casted ... and in others, can be wrapped and call existing function.
Too much duplicate.

**Comment by Brantone** [comment]
The "Details" link below points to the job run, who's console logs are: https://jenkins.ci.cloudbees.com/job/plugins/job/hockeyapp-plugin/90/console
Essentially it's running Maven on variation:
```mvn -B -f pom.xml clean install -e -U -Dignore.random.failures```

**Comment by AndreasR92** [comment]
Okay, so the build failure was not because of a compilation error, but because of findbugs finding to many violations?

Von: Brenton B <notifications@github.com>
Antworten an: jenkinsci/hockeyapp-plugin <reply@reply.github.com>
Datum: Donnerstag, 21. Juli 2016 um 09:11
An: jenkinsci/hockeyapp-plugin <hockeyapp-plugin@noreply.github.com>
Cc: Andreas Reichel <andreas.reichel@sovanta.com>, Author <author@noreply.github.com>
Betreff: Re: [jenkinsci/hockeyapp-plugin] made plugin pipeline ready by implementing SimpleBuildStep. Note that… (#31)


The "Details" link below points to the job run, who's console logs are: https://jenkins.ci.cloudbees.com/job/plugins/job/hockeyapp-plugin/90/console
Essentially it's running Maven on variation:
mvn -B -f pom.xml clean install -e -U -Dignore.random.failures

**Comment by Brantone** [comment]
Also a test failure.

**Comment by Brantone** [comment]
Do you think it's feasible to have any additional tests to support the changes?

**Comment by AndreasR92** [comment]
I don’t see any tests in the project. Where can I add my tests? Where are the 9 executed tests comming from? I think it would be very timeconsuming for me to add test coverage in a project where I have not the best insights.

Von: Brenton B <notifications@github.com>
Antworten an: jenkinsci/hockeyapp-plugin <reply@reply.github.com>
Datum: Donnerstag, 21. Juli 2016 um 18:56
An: jenkinsci/hockeyapp-plugin <hockeyapp-plugin@noreply.github.com>
Cc: Andreas Reichel <andreas.reichel@sovanta.com>, Author <author@noreply.github.com>
Betreff: Re: [jenkinsci/hockeyapp-plugin] made plugin pipeline ready by implementing SimpleBuildStep. Note that… (#31)


Do you think it's feasible to have any additional tests to support the changes?

**Comment by Brantone** [comment]
Yeah, unfortunately tests are lacking on this plugin, and I can appreciate that it's being a 3rd party external system makes it tricky to test those functions, but gotta start somewhere ... if even just a couple unit tests for the new functions.

**Comment by Brantone** [comment]
Looking for a few unit tests, and also input from community :)

**Comment by pazone** [comment]
This PR contains major functionality. Why is it still pending? What's the acceptance criteria for pull requests? Do you need some unit tests?

**Comment by Brantone** [comment]
Yup, pretty much. Something this significant would like to see some unit tests.

**Comment by 62mkv** [comment]
I've just tried out this plugin (commit 36a86e3e9dddf7bbada94476c1ba23b5e104598b), built with "mvn.bat install" on Windows 10, and uploaded on Jenkins 2.7.4 from web UI. 

Pipeline Syntax built a following snippet for me: 
`step([$class: 'HockeyappRecorder', applications: [[apiToken: '<cut out>', downloadAllowed: true, filePath: 'app/build/outputs/apk/app-qa-debug.apk', mandatory: false, notifyTeam: true, releaseNotesMethod: [$class: 'ChangelogReleaseNotes'], uploadMethod: [$class: 'VersionCreation', appId: '5af43b4cf7c047f7ad5b763fbf65553e']]], debugMode: true, failGracefully: true])`

And I get FAILURE with a following stacktrace: 

`java.lang.NullPointerException
	at hockeyapp.HockeyappRecorder.perform(HockeyappRecorder.java:181)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:69)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:59)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:52)
	at hudson.security.ACL.impersonate(ACL.java:213)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:49)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Finished: FAILURE`

what am I doing wrong ?

**Comment by Brantone** [comment]
Good reason to have some tests (although that NPE is in `perform` )
Hopefully @AndreasR92 can take a look.

**Comment by 62mkv** [comment]
BTW: if I insert a 
`currentBuild.result = 'SUCCESS'`
right before the "HockeyApp" step, "perform" executes normally

**Comment by 62mkv** [comment]
I can also confirm that, when currentBuild.result is set to 'SUCCESS', HockeyApp recorder works fine ! At least, "Upload Version" function. 

Is it normal behaviour for currentBuild.result to be NULL ?

File: src/main/java/hockeyapp/HockeyappRecorder.java [hunk 1]
```diff
@@ -144,52 +177,76 @@ private HttpClient createPreconfiguredHttpClient() {
     }
 
     @Override
+    public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath filePath, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
+        if (build.getResult().isWorseOrEqualTo(Result.FAILURE))
```
Comment (ForNeVeR) [inline src/main/java/hockeyapp/HockeyappRecorder.java]: Please add a `null` check for `build.getResult()` here.

**Comment by Brantone** [comment]
Closing, in favour of https://github.com/jenkinsci/hockeyapp-plugin/pull/32

**Comment by ynedelin** [comment]
Hi , 
I am trying to use "oldVersionHolder", delete old version option. Jenkins snippet generator returns this 
_oldVersionHolder: <object of type hockeyapp.HockeyappApplication.OldVersionHolder>,_
and no other parameters such as numberOldVersions, so it does not work.

Do you have any suggestions how to get Delete Old options work in Declarative Pipeline?

Thank you 

Yury

**Pull request closed without merging**